File: golang/1.TwoSum.go
package leetcode

func twoSum(nums []int, target int) []int {
	for i := 0; i < len(nums)-1; i++ {
		for x := i + 1; x < len(nums); x++ {
			if nums[i]+nums[x] == target {
				return []int{i, x}
			}
		}
	}

	panic("Each input has exactly one solution. This point should not be reached.")
}

File: csharp/1.TwoSum.cs
namespace Leetcode.TwoSum;

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        for (int i = 0; i < nums.Length - 1; i++) {
          for (int y = i + 1; y < nums.Length; y++) {
            if (nums[i] + nums[y] == target)
                return new[] { i, y };
          }
        }

        throw new Exception("Each input has exactly one solution.");
    }
}
